troubleshooting/tools/log-loss-test-framework/results/loss_histogram.py

```diff
@@ -0,0 +1,188 @@
+    test_name = test_run[0].strip().split()[0]
+    percent_lost = int(test_run[2].strip())
+    number_lost = int(test_run[4].strip())
+    total_input_record = test_run[6].strip()
+    success = False
+    if number_lost == 0:
+        success = True
+    tmp = test_run[10].strip()
+    split = tmp.split("=")
+    log_size = split[1]
+    tmp = test_run[11].strip()
+    split = tmp.split("=")
+    throughput = int(split[1])
+    task_id = test_run[12].strip()
+    cluster = test_run[13].strip()
+    if '/' in cluster: # some output has cluster ARN, some just name, not sure why
+        split = cluster.split("/")
+        cluster = split[1]
+    task_def = test_run[14].strip()
+    launch_type = test_run[15].strip()
+    region = test_run[16].strip()
+    buffer_size = 0
+    for pattern in TEST_NAMES_TO_BUFFER_SIZES:
+        if pattern in test_name:
+            buffer_size = TEST_NAMES_TO_BUFFER_SIZES[pattern]
+    if buffer_size == 0:
+        print("could not find buffer size in " + test_name)
+        return None
+    log_group = ""
+    total_mb = ""
+    grouped_data = test_run[9].strip()
+    kv_data = grouped_data.split()
+    for data in kv_data:
+        kv = data.split('=')
+        if kv[0] == 'group':
+            log_group = kv[1]
+        if kv[0] == 'TOTAL_SIZE_IN_MB':
+            total_mb = kv[1]
+    log_stream = os.environ.get('CW_LOG_STREAM_PREFIX', 'pre-testing-bug/logger/') + task_id
+    return {
+        'test_name': test_name,
+        'success': success,
+        'loss_%': percent_lost,
+        'number_lost': number_lost,
+        'total_input_record': total_input_record,
+        'log_size': log_size,
+        'throughput': throughput,
+        'cluster': cluster,
+        'launch_type': launch_type,
+        'buffer': buffer_size,
+        'task_id': task_id,
+        'task_def': task_def,
+        'region': region,
+        'log_group': log_group,
+        'log_stream': log_stream,
+        'total_mb': total_mb,
+    }
+
+# set this to the set of fields from above that you want to aggregate data on
+aggregate_on = ['total_input_record', 'launch_type']
+
+# aggregate data into histograms based on fields extracted from the summary
+def aggregation_key(data):
+    key = ""
+    for field in aggregate_on:
+        val = data[field]
+        key += val
+    data['key'] = key
+    return key
+
+# will be filled with loss data => the raw IDs that were not found in the log stream
+aggregated_data = {}
+
+def ingest_data(file_name):
+    file = open(file_name, 'r')
+    lines = file.readlines()
+    # string that only occurs in all summary lines
+    TEST_SUMMARY_MARKER = 'total_input_record'
+    current_data_set = []
+    key = None
+    for line in lines:
+        if TEST_SUMMARY_MARKER in line:
+            # get summary
+            index = line.find('last_lost=')
+            if index > 0:
+                index = line.find(':', index)
+                end = index - 1
+                summary = line[:end]
```

Comment:
I would expect this to be:
```
end = index
summary = line[:end]
```
Because end is not included. I suppose the last value which is the test case name from the summary section added by analyze.py is really garbage and not parsed out.